src/app/shared-services/trajectory.service.ts

```diff
@@ -70,7 +72,22 @@ export class TrajectoryService {
         )
 
       default:
-        return from(this.db.getFullTrajectory(id))
+        return new Observable<Trajectory>((subscriber) => {
+          // getting the trajectory from db and store it in cache
+          this.db.getFullTrajectory(id).then((trajectory) => {
+            this.trajectoryCache = trajectory
+
+            // publish trajectory
+            subscriber.next(this.trajectoryCache)
+
+            // subscribe to addPoint events
+            this.db.addPointSub.subscribe(async (point) => {
+              // add new point to cache trajectory and publish it
+              this.trajectoryCache.addPoint(point)
```

Comment:
Instead of cache, use `trajectory` here? (didn't test, but the closure should keep the value)

Reply:
Using the closure in b806050 which is also working (and simplifies the code even more)